micro_sam/sam_annotator/_annotator.py

```diff
@@ -0,0 +1,165 @@
+            edge_width=0.5,
+            size=12,
+            ndim=self._ndim,
+        )
+        self._point_prompt_layer.edge_color_mode = "cycle"
+
+        # Add the shape layer for box and other shape prompts.
+        self._viewer.add_shapes(
+            face_color="transparent", edge_color="green", edge_width=4, name="prompts", ndim=self._ndim,
+        )
+
+        # Add the label layers for the current object, the automatic segmentation and the committed segmentation.
+        dummy_data = np.zeros(self._shape, dtype="uint32")
+        self._viewer.add_labels(data=dummy_data, name="current_object")
+        self._viewer.add_labels(data=dummy_data, name="auto_segmentation")
+        self._viewer.add_labels(
+            data=dummy_data if segmentation_result is None else segmentation_result, name="committed_objects"
+        )
+        # Randomize colors so it is easy to see when object committed.
+        self._viewer.layers["committed_objects"].new_colormap()
+
+    def _create_widgets(self, segment_widget, segment_nd_widget, autosegment_widget, commit_widget, clear_widget):
+        self._embedding_widget = widgets.embedding_widget()
+        # Connect the call button of the embedding widget with a function
+        # that updates all relevant layers when the image changes.
+        self._embedding_widget.call_button.changed.connect(self._update_image)
+
+        self._prompt_widget = widgets.create_prompt_menu(self._point_prompt_layer, self._point_labels)
+        self._segment_widget = segment_widget()
+        widget_list = [self._embedding_widget, self._prompt_widget, self._segment_widget]
+
+        if segment_nd_widget is not None:
+            self._segment_nd_widget = segment_nd_widget()
+            widget_list.append(self._segment_nd_widget)
+
+        if autosegment_widget is not None:
+            self._autosegment_widget = autosegment_widget()
+            widget_list.append(self._autosegment_widget)
+
+        self._commit_widget = commit_widget()
+        self._clear_widget = clear_widget()
+        widget_list.extend([self._commit_widget, self._clear_widget])
+
+        # Add the widgets to the container.
+        self.extend(widget_list)
+
+    def _create_keybindings(self):
+        @self._viewer.bind_key("s")
+        def _segmet(viewer):
+            self._segment_widget(viewer)
+
+        @self._viewer.bind_key("c")
+        def _commit(viewer):
+            self._commit_widget(viewer)
+
+        @self._viewer.bind_key("t")
+        def _toggle_label(event=None):
+            vutil.toggle_label(self._point_prompt_layer)
+
+        @self._viewer.bind_key("Shift-C")
+        def _clear_annotations(viewer):
+            self._clear_widget(viewer)
+
+        if hasattr(self, "_segment_nd_widget"):
+            @self._viewer.bind_key("Shift-S")
+            def _seg_nd(viewer):
+                self._segment_nd_widget(viewer)
+
+    # TODO more clever way to integrate segmentation result so
```

Comment:
Suggest creating a dedicated issue for this suggestion, and removing the code comment here

Reply:
Yes absolutely! I will remove the TODO's and create dedicated issues before merging this PR.

Reply:
I have added dedicated issues for all the TODOs now. (I decided to leave the actual TODOs in the code and link the issue for convenience.)